cannot use _Ctype_long(p.S)  as type _Ctype_longlong

Hello, 

While trying to build the example using the latest stable version of Go and the latest version of cimgui-go the compilation fails with:

`C:\Users\ArtoS\go\pkg\mod\github.com\!allen!dang\cimgui-go@v0.0.0-20230123025400-8ac6282740cd\extra_types.go:205:25: cannot use _Ctype_long(p.S) (value of type _Ctype_long) as type _Ctype_longlong in struct literal`

In the end, I had to replace the code in the extra_types.go (on line 205) with:

`return C.ImPlotTime{S: C.longlong(p.S), Us: C.int(p.Us)}`

OS: Windows
Compiler: gcc version 12.2.0 (Rev6, Built by MSYS2 project) 64 bit

Let me know if you need more information.

--- a/cimgui_funcs.go
+++ b/cimgui_funcs.go
@@ -624,7 +624,7 @@ func (self FontGlyphRangesBuilder) Clear() {
 }
 
 func (self FontGlyphRangesBuilder) Bit(n uint64) bool {
-	return C.ImFontGlyphRangesBuilder_GetBit(self.handle(), C.xlong(n)) == C.bool(true)
+	return C.ImFontGlyphRangesBuilder_GetBit(self.handle(), C.xulong(n)) == C.bool(true)
 }
 
 func NewFontGlyphRangesBuilder() FontGlyphRangesBuilder {
@@ -632,7 +632,7 @@ func NewFontGlyphRangesBuilder() FontGlyphRangesBuilder {
 }
 
 func (self FontGlyphRangesBuilder) SetBit(n uint64) {
-	C.ImFontGlyphRangesBuilder_SetBit(self.handle(), C.xlong(n))
+	C.ImFontGlyphRangesBuilder_SetBit(self.handle(), C.xulong(n))
 }
 
 func (self FontGlyphRangesBuilder) Destroy() {
@@ -1798,7 +1798,7 @@ func DebugCheckVersionAndDataLayout(version_str string, sz_io uint64, sz_style u
 		version_strFin()
 
 	}()
-	return C.igDebugCheckVersionAndDataLayout(version_strArg, C.xlong(sz_io), C.xlong(sz_style), C.xlong(sz_vec2), C.xlong(sz_vec4), C.xlong(sz_drawvert), C.xlong(sz_drawidx)) == C.bool(true)
+	return C.igDebugCheckVersionAndDataLayout(version_strArg, C.xulong(sz_io), C.xulong(sz_style), C.xulong(sz_vec2), C.xulong(sz_vec4), C.xulong(sz_drawvert), C.xulong(sz_drawidx)) == C.bool(true)
 }
 
 func DebugTextEncoding(text string) {
@@ -3088,7 +3088,7 @@ func LoadIniSettingsFromDisk(ini_filename string) {
 // ini_size: 0
 func LoadIniSettingsFromMemoryV(ini_data string, ini_size uint64) {
 	ini_dataArg, ini_dataFin := wrapString(ini_data)
-	C.igLoadIniSettingsFromMemory(ini_dataArg, C.xlong(ini_size))
+	C.igLoadIniSettingsFromMemory(ini_dataArg, C.xulong(ini_size))
 
 	ini_dataFin()
 
@@ -3134,7 +3134,7 @@ func LogToTTYV(auto_open_depth int32) {
 }
 
 func MemAlloc(size uint64) unsafe.Pointer {
-	return unsafe.Pointer(C.igMemAlloc(C.xlong(size)))
+	return unsafe.Pointer(C.igMemAlloc(C.xulong(size)))
 }
 
 func MemFree(ptr unsafe.Pointer) {
@@ -3422,7 +3422,7 @@ func SaveIniSettingsToDisk(ini_filename string) {
 // SaveIniSettingsToMemoryV parameter default value hint:
 // out_ini_size: NULL
 func SaveIniSettingsToMemoryV(out_ini_size *uint64) string {
-	return C.GoString(C.igSaveIniSettingsToMemory((*C.xlong)(out_ini_size)))
+	return C.GoString(C.igSaveIniSettingsToMemory((*C.xulong)(out_ini_size)))
 }
 
 // SelectableBoolV parameter default value hint:
@@ -3507,7 +3507,7 @@ func SetDragDropPayloadV(typeArg string, data unsafe.Pointer, sz uint64, cond Co
 		typeArgFin()
 
 	}()
-	return C.igSetDragDropPayload(typeArgArg, (data), C.xlong(sz), C.ImGuiCond(cond)) == C.bool(true)
+	return C.igSetDragDropPayload(typeArgArg, (data), C.xulong(sz), C.ImGuiCond(cond)) == C.bool(true)
 }
 
 func SetItemAllowOverlap() {
@@ -5501,7 +5501,7 @@ func SetDragDropPayload(typeArg string, data unsafe.Pointer, sz uint64) bool {
 		typeArgFin()
 
 	}()
-	return C.wrap_igSetDragDropPayload(typeArgArg, (data), C.xlong(sz)) == C.bool(true)
+	return C.wrap_igSetDragDropPayload(typeArgArg, (data), C.xulong(sz)) == C.bool(true)
 }
 
 func SetKeyboardFocusHere() {
@@ -8375,7 +8375,7 @@ func (self ContextHook) UserData() unsafe.Pointer {
 }
 
 func (self DataTypeInfo) SetSize(v uint64) {
-	C.wrap_ImGuiDataTypeInfo_SetSize(self.handle(), C.xlong(v))
+	C.wrap_ImGuiDataTypeInfo_SetSize(self.handle(), C.xulong(v))
 }
 
 func (self DataTypeInfo) Size() float64 {

--- a/cmd/codegen/arguments_wrapper.go
+++ b/cmd/codegen/arguments_wrapper.go
@@ -30,7 +30,7 @@ func argWrapper(argType string) (wrapper argumentWrapper, err error) {
 		"const char* const[]":      charPtrPtrW,
 		"unsigned char":            simpleW("uint", "C.uchar"),
 		"unsigned char**":          uCharPtrW,
-		"size_t":                   simpleW("uint64", "C.xlong"),
+		"size_t":                   simpleW("uint64", "C.xulong"),
 		"size_t*":                  sizeTPtrW,
 		"float":                    floatW,
 		"float*":                   floatPtrW,
@@ -130,7 +130,7 @@ func uCharPtrW(arg ArgDef) ArgumentWrapperData {
 func sizeTPtrW(arg ArgDef) ArgumentWrapperData {
 	return ArgumentWrapperData{
 		ArgType: "*uint64",
-		VarName: fmt.Sprintf("(*C.xlong)(%s)", arg.Name),
+		VarName: fmt.Sprintf("(*C.xulong)(%s)", arg.Name),
 	}
 }
 

--- a/extra_types.go
+++ b/extra_types.go
@@ -2,7 +2,9 @@ package imgui
 
 // #include "cimgui_wrapper.h"
 // #include "cimplot_wrapper.h"
+// #include "extra_types.h"
 import "C"
+
 import (
 	"image/color"
 	"time"
@@ -202,7 +204,7 @@ func (i *PlotTime) fromC(p C.ImPlotTime) *PlotTime {
 }
 
 func (p PlotTime) toC() C.ImPlotTime {
-	return C.ImPlotTime{S: C.long(p.S), Us: C.int(p.Us)}
+	return C.ImPlotTime{S: C.xxlong(p.S), Us: C.int(p.Us)}
 }
 
 // wrappableType represents a GO type that can be converted into a C value

--- a/extra_types.go
+++ b/extra_types.go
@@ -204,7 +204,7 @@ func (i *PlotTime) fromC(p C.ImPlotTime) *PlotTime {
 }
 
 func (p PlotTime) toC() C.ImPlotTime {
-	return C.ImPlotTime{S: C.xxlong(p.S), Us: C.int(p.Us)}
+	return C.ImPlotTime{S: C.xlong(p.S), Us: C.int(p.Us)}
 }
 
 // wrappableType represents a GO type that can be converted into a C value

--- a/extra_types.h
+++ b/extra_types.h
@@ -2,6 +2,8 @@
 
 #ifdef __WIN32
 typedef unsigned long long xlong;
+typedef long long xlong;
 #else
 typedef unsigned long xlong;
+typedef long xxlong;
 #endif

--- a/extra_types.h
+++ b/extra_types.h
@@ -5,5 +5,5 @@ typedef unsigned long long xulong;
 typedef long long xlong;
 #else
 typedef unsigned long xulong;
-typedef long xxlong;
+typedef long xlong;
 #endif

--- a/extra_types.h
+++ b/extra_types.h
@@ -1,9 +1,9 @@
 #pragma once
 
 #ifdef __WIN32
-typedef unsigned long long xlong;
+typedef unsigned long long xulong;
 typedef long long xlong;
 #else
-typedef unsigned long xlong;
+typedef unsigned long xulong;
 typedef long xxlong;
 #endif

--- a/input_text.go
+++ b/input_text.go
@@ -72,7 +72,7 @@ func InputTextWithHint(label, hint string, buf *string, flags InputTextFlags, ca
 		labelArg,
 		hintArg,
 		(*C.char)(state.buf.ptr),
-		C.xlong(len(*buf)+1),
+		C.xulong(len(*buf)+1),
 		C.ImGuiInputTextFlags(flags),
 		C.ImGuiInputTextCallback(C.generalInputTextCallback),
 		unsafe.Pointer(&stateHandle),
@@ -101,7 +101,7 @@ func InputTextMultiline(label string, buf *string, size Vec2, flags InputTextFla
 	return C.igInputTextMultiline(
 		labelArg,
 		(*C.char)(state.buf.ptr),
-		C.xlong(len(*buf)+1),
+		C.xulong(len(*buf)+1),
 		size.toC(),
 		C.ImGuiInputTextFlags(flags),
 		C.ImGuiInputTextCallback(C.generalInputTextCallback),

